blark/transform.py

```diff
@@ -614,30 +614,43 @@ class Variable(Expression):
     ...
 
 
+@dataclass
 @_rule_handler(
     "indirection_type",
     "pointer_type",
 )
-class IndirectionType(Enum):
+class IndirectionType(Literal):
     """Indirect access through a pointer or reference."""
-    none = enum.auto()
-    pointer = enum.auto()
-    reference = enum.auto()
+    pointer_depth: int
+    reference: bool
+    meta: Optional[Meta] = meta_field()
 
     @staticmethod
-    def from_lark(token: Optional[lark.Token]) -> IndirectionType:
-        return {
-            "NONE": IndirectionType.none,
-            "POINTER TO": IndirectionType.pointer,
-            "REFERENCE TO": IndirectionType.reference,
-        }[str(token).upper()]
+    def from_lark(*tokens: Tuple[lark.Token]) -> IndirectionType:
+        pointer_depth = 0
+        reference = False
+        if len(tokens) > 0:
+            pointer_depth = tokens.count("POINTER TO")
+            reference = "REFERENCE TO" in tokens
+        return IndirectionType(
+            pointer_depth=pointer_depth,
+            reference=reference
+        )
+
+    @property
+    def value(self) -> str:
+        pointer_str = None
+        if self.pointer_depth > 0:
+            pointer_str = " ".join(["POINTER TO"] * self.pointer_depth)
+        indirection_str = join_if(
+            "REFERENCE TO" if self.reference else None,
+            " ",
+            pointer_str,
+        )
+        return indirection_str
```

Comment:
Totally unnecessary to change - as yours is equally valid! - but you could do this a bit more tersely:
`" ".join(["REFERENCE TO"] * self.reference + ["POINTER TO"] * self.pointer_depth)`

Reply:
I like this simplification! It's a good call. I spent way too much time "over-engineering" this, so I like your solution.